app/src/main/kotlin/com/twain/interprep/domain/usecase/interview/GetInterviewsUseCase.kt

```diff
@@ -0,0 +1,52 @@
+package com.twain.interprep.domain.usecase.interview
+
+import com.twain.interprep.data.model.Interview
+import com.twain.interprep.domain.repository.InterviewRepository
+import com.twain.interprep.util.InterviewsData
+import kotlinx.coroutines.flow.transform
+import java.util.Calendar
+import java.util.Date
+
+class GetInterviewsUseCase(private val interviewRepository: InterviewRepository) {
+
+    operator fun invoke() =
+        interviewRepository.getInterviews().transform { interviews ->
+            val dashBoardInterviews =
+                DashBoardInterviews(mutableListOf(), mutableListOf(), mutableListOf())
+            // To test is function, change the following line to
+            // InterviewsData.interviews.onEach
+            interviews.onEach { interview ->
+                if (interview.date.before(Date())) {
+                    dashBoardInterviews.pastInterviews.add(interview)
+                } else if (isSameWeekAsCurrentDate(interview.date)) {
+                    dashBoardInterviews.upcomingInterviews.add(interview)
+                } else {
+                    dashBoardInterviews.comingNextInterviews.add(interview)
+                }
+            }
+            emit(dashBoardInterviews)
+        }
+
+
+    private fun isSameWeekAsCurrentDate(date: Date): Boolean {
```

Comment:
nit: Naming suggestion
```suggestion
    private fun isInterviewDateInSameWeek(date: Date): Boolean {
```